onnxruntime/core/providers/webgpu/nn/layer_norm.cc

```diff
@@ -24,7 +24,13 @@ Status LayerNormProgram::GenerateShaderCode(ShaderHelper& shader) const {
   if (has_bias_) {
     shader.AddInput("bias", ShaderUsage::UseUniform);
   }
-  shader.AddOutput("output", ShaderUsage::UseUniform);
+  shader.AddOutput("y", ShaderUsage::UseUniform);
+  if (output_count_ > 1) {
+    shader.AddOutput("mean_output", ShaderUsage::None);
```

Comment:
These two outputs of `Mean` and `InvStdDev` are all optional, if one model has two outputs, which may be [`y`, `mean`] or [`y`, `InvStdDev`]? Is it reasonable to define two variables `has_mean_` and `has_inv_std_dev`?
Refer https://github.com/microsoft/onnxruntime/blob/64b0d071a7ce4f42495b663aec993d3e12fbc5d8/onnxruntime/contrib_ops/webgpu/bert/skip_layer_norm.h#L24

Reply:
Although the JS op uses `output_count`, I think your point makes more sense.